### DISCUSSION TITLE
Can I generate diagram images from command line?

### DISCUSSION THREAD

**Discussion by renannprado** [structurizr/lite#58] [Q&A] [answered]
### Question

I'm not sure if this is the right place to ask, but how can one run the docker container in an automated way to generate the diagrams as they are displayed on the web? Since I need to include some diagrams in the documentation, it would be nice if I could automate the generation of the files so that they land in the places where they belong in the folder structure rather then opening it, downloading the images, etc...

Edit:

And no, I don't want to use PlantUML because the output is not the same.

**Comment by simonbrowndotje** [comment, marked as answer]
Yes, you can use headless Chrome and Puppeteer to automate diagram exports; see https://github.com/structurizr/puppeteer for more.

**Reply by renannprado** [reply]
@simonbrowndotje I'm marking this as an answer, but I would have expected the tool itself (structurizr) to be able to output the diagrams from the CLI. Is there any reason why this is not supported?

**Reply by simonbrowndotje** [reply]
> Is there any reason why this is not supported?

The CLI is written in Java, whereas Node is required to run headless Chrome/Puppeteer.

> but I would have expected the tool itself (structurizr) to be able to output the diagrams from the CLI.

You're welcome to submit a PR. 🙂
